lib/beautiful/gtk.lua

```diff
@@ -0,0 +1,171 @@
+---------------------------------------------------------------------------
+--- Querying current GTK+ 3 theme via GtkStyleContext.
+--
+-- @author Yauheni Kirylau &lt;yawghen@gmail.com&gt;
+-- @copyright 2016-2017 Yauheni Kirylau
+-- @module beautiful.gtk
+---------------------------------------------------------------------------
+local gears_debug = require("gears.debug")
+local gears_math = require("gears.math")
+local join = require("gears.table").join
+local unpack = unpack or table.unpack -- luacheck: globals unpack (compatibility with Lua 5.1)
+
+
+local gtk = {
+  cached_theme_variables = nil
+}
+
+
+local function convert_gtk_channel_to_hex(channel_value)
+    return string.format("%02x", gears_math.round(channel_value * 255))
+end
+
+local function convert_gtk_color_to_hex(gtk_color)
+  return "#" ..
+    convert_gtk_channel_to_hex(gtk_color.red) ..
+    convert_gtk_channel_to_hex(gtk_color.green) ..
+    convert_gtk_channel_to_hex(gtk_color.blue) --..
+  --convert_gtk_channel_to_hex(gtk_color.alpha)
```

Comment:
btw  pixmap or i commented it out because tasklist and taglist were not working properly with such colors, i'll try mb after https://github.com/awesomeWM/awesome/pull/2127 is merged it will work fine

Reply:
Uhm, which colors do you mean exactly? The background color uses a `wibox.container.background`, but the foreground color uses Pango markup (`<span font_desc='foo'><span color='bar'>etc`) and Pango... uhm, should support alpha, too, according to https://developer.gnome.org/pango/unstable/PangoMarkupFormat.html ...? This seems like something that we should text&fix.

Reply:
weird, could be some `--no-rgba`-related kind of thing -- i've just tried it on other machine and it works as expected

Reply:
so i am now on the workstation with radeon again, the issue is still reproduciable and happens only to taglist and tasklist `fg` color, `#888a85ff` is rendered as white and `#acafa9ff` as black

Reply:
found the problem:
```
$ awesome-client 'return require("gears.color").ensure_pango_color("#888a85")'
   string "#888a85"

$ awesome-client 'return require("gears.color").ensure_pango_color("#888a85ff")'
   string "black"

```

Reply:
Uhm, so, why does this work on one computer but not the other?
Anyway, [PangoColor](https://developer.gnome.org/pango/unstable/pango-Text-Attributes.html#PangoColor-struct) has just red/green/blue and [alpha](https://developer.gnome.org/pango/unstable/pango-Text-Attributes.html#pango-color-parse) is [implemented](https://developer.gnome.org/pango/unstable/pango-Text-Attributes.html#pango-attr-background-alpha-new) via two other, special properties. Pango markup hides that difference, but `pango_color_parse` can, of course, not. Meh...